client/src/dropboxClient.cpp

```diff
@@ -1,50 +1,96 @@
-#include "../../util/include/dropboxUtil.hpp"
-#include "../../util/include/File.hpp"
-#include "../include/dropboxClient.hpp"
-
 #include <iostream>
 #include <cstring>
-#include <string>
 #include <stdexcept>
 
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/socket.h>
-#include <netinet/in.h>
 #include <arpa/inet.h>
 #include <netdb.h>
+#include <pwd.h>
+
+#include <boost/filesystem.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include "../../util/include/string_formatter.hpp"
+#include "../../util/include/table_printer.hpp"
+#include "../../util/include/File.hpp"
+#include "../include/dropboxClient.hpp"
+
+#include "../include/login_command_parser.hpp"
 
 const std::string Client::LOGGER_NAME = "Client";
 
-Client::Client(uint64_t device_id, const std::string &user_id) : device_id_(device_id), user_id_(user_id) {
+Client::Client()
+{
     logger_ = spdlog::stdout_color_mt(LOGGER_NAME);
     logger_->set_level(spdlog::level::debug);
 }
 
-Client::~Client() {
+Client::~Client()
+{
     spdlog::drop(LOGGER_NAME);
 }
 
-void Client::login_server(const std::string& host, int32_t port) {
+void Client::start_client(int argc, char **argv)
+{
+    if (logged_in_)
+        return;
+
+    LoginCommandParser login_command_parser;
+
+    login_command_parser.ParseInput(argc, argv);
+
+    if (login_command_parser.ShowHelpMessage())
```

Comment:
O que ShowHelpMessage faz pra servir como condição que decide se o programa encerra?

Reply:
Em geral quando tu inicia um programa e pede a mensagem de ajuda tu quer só a mensagem de ajuda pra saber como usar o programa, então aqui se a mensagem for mostrada não tem mais nada pra fazer